### ISSUE TITLE
Add flag to report unneeded stylelint-disable comments

### ISSUE THREAD

**Issue by jonrohan** [stylelint/stylelint#1184]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

This is a feature request. I want the ability to ignore the `stylelint-disable` comments in the source files from the command line. This might sound crazy at first but the reason I want to do this is so I can test when I have a disabled comment and don't need it.

I run a script to get a list of linters disabled in comments, then run stylelint on those files without the disable comments and see if the error names I get match up with what's expected.

Breaking the CI build when code was refactored enough to remove the disable comment, but the comment wasn't removed. 

**Why would I do this?** Because having lingering disabled comments in code are openings where more bad code can get it. If you're adding stylelint to an existing huge codebase, you want to try and fix as much as you can but disable where you can't yet, or don't have the bandwidth to. Coming back and refactoring later.

> Which rule, if any, is this issue related to?

This is related to the interface CLI, Node, Grunt, etc..

> What CSS is needed to reproduce this issue?

```css
/* stylelint-disable selector-no-id */
.foo {  /* <--- selector-no-id not needed to be disabled here */
  color: pink;
}
```

> Which version of stylelint are you using?

`6.2.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"CLI with `bin/stylelint "app/assets/stylesheets/**/*.scss" --config app/assets/stylesheets/.stylelintrc --syntax scss`"

> Alternative goal 🏆 

What I'm trying to accomplish is not related to the issue title, so if you think this should be a plugin, or is possible as a plugin, I'd love some ideas/suggestions. I'm pretty new to stylelint and loving what i'm seeing so far.

**Comment by jeddy3** [comment]
@jonrohan Thanks for the suggestion.

> I want to do this is so I can test when I have a disabled comment and don't need it.

That's a valid use-case and a very interesting one :)

> I run a script to get a list of linters disabled in comments, *then run stylelint on those files without the disable comments* and see if the error names I get match up with what's expected.

So, how are you doing this at the moment? Is your script removing the disable comments from (a copy of) your source files, before running stylelint again them? And so, what you're looking for is a more convenient way ignoring the disable comments, rather than having to duplicate the source code and remove the comments on each CI build?

**Comment by jonrohan** [comment]
> So, how are you doing this at the moment? Is your script removing the disable comments from (a copy of) your source files, before running stylelint again them? 

I should preface this whole discussion with these are rails tests, so have to be written in ruby.

Yes currently my approach is to grep for a list of files containing `stylelint-disable` and create a hash with the path and the list of unique disabled lint rules. Then I loop over that list, reading the file body. I replace any instances of `stylelint-disable` with `` in the body and pipe each file separately to CLI stylelint. 

With aprox. 100 files this is a pretty slow test. What I think would make it faster is to get the list of files and disabled rules, then passing a glob of those files to `stylelint` running it once (with inline disabled comments turned off). 

```
stylelint "dir/file1.scss dir/file2.scss dir/file2.scss" --syntax scss --ignore-disabled-rules -f json
```

And compare the results. 

> If so, is what you're looking for a more convenient way of ignoring the disable comments, rather than having to duplicate the source code and remove the comments on each CI build?

Yes, I think just having the ability to flip it off from the CLI would let me run all my files with disabled rules against the linter.

**Comment by jeddy3** [comment]
Thanks for the explanation. A `--ignore-disabled-rules` option sounds like a good addition to me. Although, we might want to call it `--ignore-disable-comments`.

@davidtheclark Do you see any issues with this proposal? If not, I'll label this as "help wanted".

**Comment by davidtheclark** [comment]
Yeah, sounds fine. I like `ignore-disable-comments`.

**Comment by jeddy3** [comment]
> Yeah, sounds fine. I like ignore-disable-comments.

Good stuff.

@jonrohan If you'd like to help contribute this feature, please feel free to fire over a Pull Request.

**Comment by jonrohan** [comment]
> If you'd like to help contribute this feature, please feel free to fire over a Pull Request.

I'll give it a shot.

**Comment by jeddy3** [comment]
This feature is now about *reporting* unneeded `stylelint-disable` comments

**Comment by davidtheclark** [comment]
Closed by #1750.
